pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -50,6 +50,8 @@ func (p *Pipeline) Validate(ctx context.Context) *apis.FieldError {
 	// When a Pipeline is created directly, instead of declared inline in a PipelineRun,
 	// we do not support propagated parameters and workspaces.
 	// Validate that all params and workspaces it uses are declared.
+	errs = errs.Also(PipelineTaskList(p.Spec.Tasks).validateUsageOfDeclaredPipelineTaskParameters(ctx, "tasks"))
+	errs = errs.Also(PipelineTaskList(p.Spec.Finally).validateUsageOfDeclaredPipelineTaskParameters(ctx, "finally"))
```

Comment:
Can we move this into `validatePipelineParameterUsage` instead?

Reply:
Yes! Thanks for the suggestion.